File: C/pset2/readability/readability.c
/**

  * Readability

  * M. Ashuza

  * A program that computes the approximate grade level needed.

  **/

#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <math.h>

int main(void)
{
    string s = get_string("Text: ");
    int num_of_letters = 0;
    int num_of_word = 1;
    int num_of_sentences = 0;

// Count number of word sentences and letters
    for (int i = 0; i < strlen(s); i++)
    {
        if ((s[i] >= 'a' && s[i] <= 'z') || (s[i] >= 'A' && s[i] <= 'Z'))
        {
            num_of_letters++;
        }
        else if (s[i] == ' ')
        {
            num_of_word++;
        }
        else if (s[i] == '.' || s[i] == '!' || s[i] == '?')
        {
            num_of_sentences++;
        }
    }
// Calculate using the Coleman-Liau index
    float grade = 0.0588 * (100 * (float) num_of_letters / (float) num_of_word) - 0.296 * (100 * (float) num_of_sentences
                  / (float) num_of_word) - 15.8;
// Print the grade
    if (grade < 16 && grade >= 0)
    {
        printf("Grade %i\n", (int) round(grade));
    }
    else if (grade >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Before Grade 1\n");
    }

}

File: python/pset6/readability/readability.py
###

# credit.py

# M. Ashuza

# A program for validating credit card numbers.

###
from cs50 import get_string

num_words, num_letters, num_sentences = 0, 0, 0
# Get user input
s = get_string("Text: ").strip()  # " progarmming " -> "progarmming"

# Iterate over every single letter
# to count num of letter, words and sentences
for i in range(len(s)):
    if (i == 0 and s[i] != " ") or (i != len(s) - 1 and s[i] == " " and s[i + 1] != " "):
        num_words += 1
    if s[i].isalpha():
        num_letters += 1
    if s[i] == "." or s[i] == "!" or s[i] == "?":
        num_sentences += 1

# Calculate using the Coleman-Liau index
L = num_letters / num_words * 100
S = num_sentences / num_words * 100
index = round(0.0588 * L - 0.296 * S - 15.8)

# Print the grade level
if index < 1:
    print("Before Grade 1")
elif index >= 16:
    print("Grade 16+")
else:
    print(f"Grade {index}")

